frontend/pages/hosts/details/cards/Software/SelfService/SelfService.tsx

```diff
@@ -80,14 +97,75 @@ const SoftwareSelfService = ({
   onShowInstallDetails,
   onShowUninstallDetails,
 }: ISoftwareSelfServiceProps) => {
-  const { renderFlash } = useContext(NotificationContext);
+  const { renderFlash, renderMultiFlash } = useContext(NotificationContext);
 
   const [selfServiceData, setSelfServiceData] = useState<
     IGetDeviceSoftwareResponse | undefined
   >(undefined);
   const [showUninstallSoftwareModal, setShowUninstallSoftwareModal] = useState(
     false
   );
+  const [updatesPage, setUpdatesPage] = useState(0);
+  const [updatesPageSize, setUpdatesPageSize] = useState(() =>
+    getUpdatesPageSize(window.innerWidth)
+  );
+
+  const enhancedSoftware = useMemo(() => {
+    if (!selfServiceData) return [];
+    return selfServiceData.software.map((software) => ({
+      ...software,
+      ui_status: getUiStatus(software, true),
+    }));
+  }, [selfServiceData]);
+
+  const updateSoftware = enhancedSoftware.filter(
+    (software) =>
+      software.ui_status === "updating" ||
+      software.ui_status === "pending_update" || // Should never show as self-service = host online
+      software.ui_status === "update_available" ||
+      software.ui_status === "failed_install_update_available" ||
+      software.ui_status === "failed_uninstall_update_available"
+  );
+
+  useEffect(() => {
+    const handleResize = () => {
+      const newPageSize = getUpdatesPageSize(window.innerWidth);
+      setUpdatesPageSize(() => {
+        const newTotalPages = Math.ceil(updateSoftware.length / newPageSize);
+        setUpdatesPage((prevPage) => {
+          // If the current page is now out of range, go to the last valid page
+          return Math.min(prevPage, Math.max(0, newTotalPages - 1));
+        });
+        return newPageSize;
+      });
+    };
+
+    window.addEventListener("resize", handleResize);
```

Comment:
it may or may not be an issue but `handleResize` is going to be called on every resize event, which means `setUpdatesPageSize` will be called on every resize event, which means this component will rerender on all resize events. if this looks to be a problem then we should think about debouncing the handler function.